### PR TITLE
feat(zk_toolbox): Add installation script

### PR DESCRIPTION
**Pull request by matias-gonz**
## What ❔
Add installation script

Readme visualization:
https://github.com/matter-labs/zksync-era/blob/matias-zkup/zk_toolbox/zkup/README.md

### CODE REVIEW

File: zk_toolbox/zkup/zkup [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+ensure() {
+  if ! "$@" >/dev/null 2>&1; then
+    err "command failed: $*"
+    exit 1
+  fi
+}
+
+ensure_spinner() {
+  local cmd="$*"
+  "$@" >/dev/null 2>&1 &
```
Comment (Deniallugo) [inline zk_toolbox/zkup/zkup:175]: i don't think we have to send everything to dev null during the installation

File: zk_toolbox/zkup/zkup [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+ZKT_BIN_DIR="$ZKT_DIR/bin"
+
+ZKUP_SKIP_ZK_SUPERVISOR=0
+
+BINS=(zk_inception zk_supervisor)
+
+main() {
+  parse_args "$@"
+
+  say "Installing zk_toolbox"
```
Comment (Deniallugo) [inline zk_toolbox/zkup/zkup:15]: can we add some info what and from what we are installing? 
```
Repo       : https://github.com/foundry-rs/
Book       : https://book.getfoundry.sh/

```
Like this?

File: zk_toolbox/zkup/zkup [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+  check_prerequisites
+  mkdir -p "$ZKT_BIN_DIR"
+
+  if [ -n "$ZKUP_PATH" ]; then
+    install_local
+  else
+    install_from_repo
+  fi
+
+  success "zk_toolbox installed successfully"
```
Comment (Deniallugo) [inline zk_toolbox/zkup/zkup:25]: let's print what exactly has been installed

File: zk_toolbox/zkup/zkup (hunk 2 above)
Comment (Deniallugo) [reply zk_toolbox/zkup/zkup:15]: And let's add some info about zk toolbox here as well

File: zk_toolbox/zkup/zkup (hunk 1 above)
Comment (matias-gonz) [reply zk_toolbox/zkup/zkup:175]: Okay, I will remove the spinner since it wouldn't work

**Review by Deniallugo — DISMISSED** [review]

**Review by yorik — DISMISSED** [review]

**Review by Deniallugo — DISMISSED** [review]

**Review by Romsters — APPROVED** [review]

**Review by Deniallugo — APPROVED** [review]

**Pull request merged**